buffer/src/main/java/io/netty/buffer/api/pool/PoolThreadCache.java

```diff
@@ -59,27 +59,28 @@
                     int freeSweepAllocationThreshold) {
         checkPositiveOrZero(maxCachedBufferCapacity, "maxCachedBufferCapacity");
         this.freeSweepAllocationThreshold = freeSweepAllocationThreshold;
-        this.arena = arena;
         if (arena != null) {
             // Create the caches for the heap allocations
-            smallSubPageCaches = createSubPageCaches(
+            MemoryRegionCache[] smallSubPageCaches = createSubPageCaches(
                     smallCacheSize, arena.numSmallSubpagePools);
 
-            normalCaches = createNormalCaches(
+            MemoryRegionCache[] normalCaches = createNormalCaches(
                     normalCacheSize, maxCachedBufferCapacity, arena);
 
-            arena.numThreadCaches.getAndIncrement();
+            // Only check if there are caches in use.
+            if ((smallSubPageCaches != null || normalCaches != null)
+                && freeSweepAllocationThreshold < 1) {
+                throw new IllegalArgumentException("freeSweepAllocationThreshold: "
+                                                   + freeSweepAllocationThreshold + " (expected: > 0)");
+            }
+
+            cacheRef = new WeakReference<>(new Cache(arena, smallSubPageCaches, normalCaches));
+            arenaReferenceCounter = arena.numThreadCaches;
+            arenaReferenceCounter.getAndIncrement();
```

Comment:
Is this change related ? Seems like no.

Reply:
This is from #11710, which this PR is based on.